File: frontend/src/services/eventTypeService.js
import axiosClient from '../api/axiosClient';

/**
 * Event Types API Service wrapper.
 */
export const eventTypeService = {
  // Public
  getActiveEventTypes: async () => {
    const res = await axiosClient.get('/event-types');
    return res.data;
  },

  // Admin
  getAllEventTypes: async () => {
    const res = await axiosClient.get('/event-types/all');
    return res.data;
  },

  getById: async (id) => {
    const res = await axiosClient.get(`/event-types/${id}`);
    return res.data;
  },

  create: async (data) => {
    const res = await axiosClient.post('/event-types', data);
    return res.data;
  },

  update: async (id, data) => {
    const res = await axiosClient.put(`/event-types/${id}`, data);
    return res.data;
  },

  delete: async (id) => {
    const res = await axiosClient.delete(`/event-types/${id}`);
    return res.data;
  },
};

File: backend/src/main/java/com/eventeasy/service/EventTypeService.java
package com.eventeasy.service;

import com.eventeasy.dto.request.EventTypeRequest;
import com.eventeasy.dto.response.EventTypeResponse;

import java.util.List;
import java.util.UUID;

/**
 * <p>Service interface declaring EventType management operations.</p>
 *
 * @author EventEasy Engineering
 * @version 1.0.0
 */
public interface EventTypeService {

    /**
     * Retrieve list of all active public event types.
     *
     * @return List of EventTypeResponse DTOs
     */
    List<EventTypeResponse> getActiveEventTypes();

    /**
     * Retrieve all non-deleted event types for admin management.
     *
     * @return List of EventTypeResponse DTOs
     */
    List<EventTypeResponse> getAllEventTypes();

    /**
     * Retrieve EventType by UUID identifier.
     *
     * @param id event type UUID
     * @return EventTypeResponse DTO
     */
    EventTypeResponse getEventTypeById(UUID id);

    /**
     * Create new EventType.
     *
     * @param request creation payload
     * @return created EventTypeResponse DTO
     */
    EventTypeResponse createEventType(EventTypeRequest request);

    /**
     * Update existing EventType.
     *
     * @param id event type UUID
     * @param request update payload
     * @return updated EventTypeResponse DTO
     */
    EventTypeResponse updateEventType(UUID id, EventTypeRequest request);

    /**
     * Soft delete EventType record.
     *
     * @param id event type UUID
     */
    void deleteEventType(UUID id);
}
